clang/test/CodeGen/arm-vector_type-params-returns.c

```diff
@@ -27,7 +27,7 @@
 #endif
 
 // function return types
-// CHECK-LABEL: define dso_local <8 x half> @test_ret_v8f16(
+// CHECK-LABEL: define dso_local noundef <8 x half> @test_ret_v8f16(
```

Comment:
Do you know why clang places the noundef on the argument but not the return for these, despite them having the same type?

Reply:
Ah, it seems to indeed be a "weird C rule":
https://github.com/llvm/llvm-project/blob/3c92011b600bdf70424e2547594dd461fe411a41/clang/lib/CodeGen/CGCall.cpp#L1816-L1819